File: src/Tests.fs/HellgateTests.fs
module TLK.PortalsAndHeroes.Tests.Hellgate

open NUnit.Framework
open FsUnit.TopLevelOperators
open TLK.PortalsAndHeroes.Kernel

let inscription (hellgate : Hellgate) = hellgate.Inscription
let close (hellgate : Hellgate) = hellgate.Close(); hellgate

[<Test>]
let ``hellgate inscription should be "Lasciate ogni speranza, voi ch'entrate"`` () =
    () |> Hellgate |> inscription |> should equal "Lasciate ogni speranza, voi ch'entrate"

[<Test>]
let ``closed hellgate inscription should be empty`` () =
    () |> Hellgate |> close |> inscription |> should be EmptyString

File: src/Tests.cs/HellgateTests.cs
using FluentAssertions;
using NUnit.Framework;
using TLK.PortalsAndHeroes.Kernel;

namespace TLK.PortalsAndHeroes.Tests
{
    internal class HellgateTests
    {
        private readonly Hellgate hellgate = new Hellgate();

        [Test]
        public void HellgateInscriptionShouldBeLasciateOgniSperanzaVoiChentrate() =>
            hellgate.Inscription.Should().Be("Lasciate ogni speranza, voi ch'entrate");
    }
}
